<launch> 
    <arg name="cam_id" default="0"/>
    <arg name="wired_sync_mode" value="$(eval 1 if int(arg('cam_id')) == 0 else 2)"/>
    <arg name="sensor_sn" value="$(eval '000003493812,000180921812,000059793712,000263392612'.split(',')[arg('cam_id')])"/>

  <group ns="k4a_$(arg cam_id)">
    <include file="$(find rpad_ros)/launch/launch_cam.launch">
        <arg name="sensor_sn" value="$(arg sensor_sn)"/>
        <arg name="camera_id" value="$(arg cam_id)"/>
        <arg name="wired_sync_mode" value="$(arg wired_sync_mode)"/>
    </include>

    <!-- Publish transforms -->
    <include file="$(find rpad_ros)/launch/publish_calibration.launch">
      <arg name="namespace_prefix" value="k4a_$(arg cam_id)"/>
      <arg name="tracking_base_frame" value="k4a_$(arg cam_id)/camera_base"/>
    </include>
  </group>

</launch>